pkg/util/ipset/ipset.go

```diff
@@ -72,74 +72,148 @@ var EntryMemberPattern = "(?m)^(.*\n)*Members:\n"
 // ipset version output is similar to "v6.10".
 var VersionPattern = "v[0-9]+\\.[0-9]+"
 
+// Option is variadic initializer for IPSet
+type Option func(set *IPSet)
+
+// NewIPSet returns a new IPSet struct
+func NewIPSet(name string, setType Type, opts ...Option) (IPSet, error) {
```

Comment:
I'm not sure the `Option` thing is really needed... there's only a single "real" (ie, non-test) caller of this constructor. You could just make the constructor take exactly the args that that caller uses.

Reply:
I think this was my fault - I was looking at it as an API design problem.  The set of arguments needed is dependent on the setType.